File: dtos/GuestCreationOptions.cs
using System;
using System.ComponentModel.DataAnnotations;

namespace HealthGyro.Models.Dtos
{
   public class GuestCreationOptionsDto
   {
      [Required, DataType(DataType.EmailAddress)]
      public string Email { get; set; }
      [Required]
      public string FirstName { get; set; }
      [Required]
      public string LastName { get; set; }
      public DateTime? AccessExpiresAt { get; set; }
      public bool CanAccessFinances { get; set; }
      public bool CanAccessMedication { get; set; }
      public bool CanAccessAppointments { get; set; }
      public bool CanAccessLabResults { get; set; }
   }
}

File: schemas/validation/guest-creation-options.ts
import {getEmailValidationMessage, getRequiredValidationMessage} from './utils';
import {object, string} from 'yup';

const guestCreationOptions = object({
    email: string().email(getEmailValidationMessage).required(getRequiredValidationMessage),
    firstName: string().required(getRequiredValidationMessage),
    lastName: string().required(getRequiredValidationMessage),
});

export default guestCreationOptions;